src/json.ts

```diff
@@ -0,0 +1,139 @@
+import deepEqual from 'fast-deep-equal';
+import { hasProperty } from './misc';
+
+/**
+ * Any JSON-compatible value.
+ */
+export type Json =
+  | null
+  | boolean
+  | number
+  | string
+  | Json[]
+  | { [prop: string]: Json };
```

Comment:
Maybe use `Record<string, Json>` instead

Reply:
IIRC there's a reason we didn't do it that way cc: @Gudahtt